src/Classes/PartyTab.lua

```diff
@@ -0,0 +1,620 @@
+-- Path of Building
+--
+-- Module: Party Tab
+-- Party tab for the current build.
+--
+local pairs = pairs
+local ipairs = ipairs
+local s_format = string.format
+local t_insert = table.insert
+
+local PartyTabClass = newClass("PartyTab", "ControlHost", "Control", function(self, build)
+	self.ControlHost()
+	self.Control()
+
+	self.build = build
+	
+	self.processedInput = { Aura = {}, Curse = {} }
+	self.enemyModList = new("ModList")
+	self.buffExports = {}
+	self.enableExportBuffs = true
+
+	self.lastContentAura = ""
+	self.lastContentCurse = ""
+	self.lastContentEnemyCond = ""
+	self.lastContentEnemyMods = ""
+	self.lastEnableExportBuffs = true
+	self.showColorCodes = false
+
+	local notesDesc = [[^7DO NOT EDIT ANY BOXES UNLESS YOU KNOW WHAT YOU ARE DOING, use copy/paste instead, or import
+	To import a build, you must export that build with "Export Support" ticked
+	The Strongest Aura applies, but your curses override the supports regardless of strength
+	]]
+	
+	self.controls.notesDesc = new("LabelControl", {"TOPLEFT",self,"TOPLEFT"}, 8, 8, 150, 16, notesDesc)
+	self.controls.notesDesc.width = function()
+		local width = self.width / 2 - 16
+		if width ~= self.controls.notesDesc.lastWidth then
+			self.controls.notesDesc.lastWidth = width
+			self.controls.notesDesc.label = table.concat(main:WrapString(notesDesc, 16, width - 50), "\n")
+		end
+		return width
+	end
+	self.controls.importCodeHeader = new("LabelControl", {"TOPLEFT",self.controls.notesDesc,"BOTTOMLEFT"}, 0, 32, 0, 16, "^7Enter a build code below: (NOT URL)")
+	self.controls.importCodeHeader.y = function()
+		local lineCount = 1
+		for i = 1, #self.controls.notesDesc.label do
+			local c = self.controls.notesDesc.label:sub(i, i)
+			if c == '\n' then lineCount = lineCount + 1 end
+		end
+
+		return (lineCount - 2) * 16
+	end
+	
+	local importCodeHandle = function (buf)
```

Comment:
This function should be pulled out into a shared space between the import tab and this

Reply:
I mentioned this in discord, but the functions reference to self, and to specific controls makes its "internal lua object" very different and very little of it can actually be de-duplicated, which is why I didnt do it originally was I was trying to find a way to share it between the 2